src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3436 @@
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 14, 62, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_An/func_80B57FC4.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+s32 EnAn_InitializeObjectSlots(EnAn* this, PlayState* play) {
+    s32 ret = false;
+
+    if ((this->roomNum != play->roomCtx.curRoom.num) && (play->roomCtx.status == 0) && !this->slotsInitialized) {
+        this->msmoObjectSlot = SubS_GetObjectSlot(OBJECT_MSMO, play);
+        this->an4ObjectSlot = SubS_GetObjectSlot(OBJECT_AN4, play);
+        this->maskKerfayObjectSlot = SubS_GetObjectSlot(OBJECT_MASK_KERFAY, play);
+        this->an3ObjectSlot = SubS_GetObjectSlot(OBJECT_AN3, play);
+        this->an2ObjectSlot = SubS_GetObjectSlot(OBJECT_AN2, play);
+        this->actor.draw = NULL;
+        this->roomNum = play->roomCtx.curRoom.num;
+        this->slotsInitialized = true;
+    }
+
+    if (!this->slotsInitialized) {
+        return false;
+    }
+
+    if (((this->an2ObjectSlot > OBJECT_SLOT_NONE) && !SubS_IsObjectLoaded(this->an2ObjectSlot, play)) ||
+        ((this->an3ObjectSlot > OBJECT_SLOT_NONE) && !SubS_IsObjectLoaded(this->an3ObjectSlot, play)) ||
+        ((this->maskKerfayObjectSlot > OBJECT_SLOT_NONE) && !SubS_IsObjectLoaded(this->maskKerfayObjectSlot, play)) ||
+        ((this->an4ObjectSlot > OBJECT_SLOT_NONE) && !SubS_IsObjectLoaded(this->an4ObjectSlot, play)) ||
+        ((this->msmoObjectSlot > OBJECT_SLOT_NONE) && !SubS_IsObjectLoaded(this->msmoObjectSlot, play))) {
+        ret = true;
+    } else {
+        this->actor.draw = EnAn_Draw;
+        this->slotsInitialized = false;
+    }
+
+    return ret;
+}
+
+Actor* EnAn_FindActor(EnAn* this, PlayState* play, u8 actorCategory, s16 actorId) {
+    Actor* foundActor = NULL;
+
+    while (true) {
+        foundActor = SubS_FindActor(play, foundActor, actorCategory, actorId);
+
+        if (foundActor == NULL) {
+            break;
+        }
+
+        if ((this != (EnAn*)foundActor) && (foundActor->update != NULL)) {
+            break;
+        }
+
+        if (foundActor->next == NULL) {
+            foundActor = NULL;
+            break;
+        }
+
+        foundActor = foundActor->next;
+    }
+
+    return foundActor;
+}
+
+// Name after ENAN_8000
+Actor* func_80B53A7C(EnAn* this, PlayState* play, u8 actorCategory, s16 actorId) {
+    Actor* foundActor = NULL;
+
+    while (true) {
+        foundActor = SubS_FindActor(play, foundActor, actorCategory, actorId);
+
+        if (foundActor == NULL) {
+            break;
+        }
+
+        if ((this != (EnAn*)foundActor) && (foundActor->update != NULL)) {
+            if (!(foundActor->params & ENAN_8000)) {
```

Comment:
Could also use the macro here, just not 100% sure since the function can be used for other actors (though it only really makes sense for Anju and only gets used with Anju).